src/Simplex/Messaging/Agent.hs

```diff
@@ -433,10 +588,11 @@ subscribeConnection' c connId =
 subscribeConnections' :: forall m. AgentMonad m => AgentClient -> [ConnId] -> m (Map ConnId (Either AgentErrorType ()))
 subscribeConnections' _ [] = pure M.empty
 subscribeConnections' c connIds = do
-  conns :: Map ConnId (Either StoreError SomeConn) <- M.fromList . zip connIds <$> withStore' c (forM connIds . getConn)
-  let (errs, cs) = M.mapEither id conns
+  conns :: [(ConnId, Either StoreError SomeConn)] <- zip connIds <$> withStore' c (forM connIds . getConn)
```

Comment:
not sure why this change was needed?

Reply:
what would be better is to have rcvOrSndQueue return `Maybe (Either (SndQueue, ConnData) (RcvQueue, ConnData))` and then partition Nothings away with mapMaybe, rather than filtering this list multiple times.